projects/components/src/table/data/table-cdk-data-source.ts

```diff
@@ -59,23 +62,45 @@ export class TableCdkDataSource implements DataSource<TableRow> {
 
     return this.rowsChange$.pipe(
       tap(rows => this.cacheRows(rows)),
-      tap(rows => this.loadingState.next(of(rows)))
+      tap(rows => this.cacheValues(rows)),
+      tap(rows => this.loadingStateSubject.next(of(rows)))
     );
   }
 
   public disconnect(): void {
     this.rowsChange$.complete();
-    this.loadingState.complete();
+    this.loadingStateSubject.complete();
   }
 
   public getValues(columnConfig: TableColumnConfig): unknown[] {
-    return [...new Set(this.cachedRows.map(row => row[columnConfig.field]))].sort(sortUnknown);
+    return this.cachedValues.has(columnConfig.field)
+      ? [...this.cachedValues.get(columnConfig.field)!].sort(sortUnknown)
```

Comment:
Why not sort on set rather than every get?

Reply:
Smart man. Good catch!

Reply:
Ah, I actually remember now. This is a set so its unsorted.

What I can do instead though is use a set to build the unique list but store an array in the map instead.

Reply:
Easy confusion, but JS Sets are actually sorted according to the spec, so up to you.
> Set objects are collections of values. You can iterate through the elements of a set in insertion order. A value in the Set may only occur once; it is unique in the Set's collection.

Reply:
Correct, but they can't be sorted in place. I'd still have to sort prior to adding all of the values to the set. So same solution of building an array and sorting. It just changes when.

I'll build the set first to collect only unique, then transfer to `cachedValues` as a sorted array.